app/assets/javascripts/school_administrator_dashboard/dashboard_components/absences_dashboard/SchoolwideAbsences.js

```diff
@@ -3,24 +3,48 @@ import PropTypes from 'prop-types';
 
 import DashboardHelpers from '../DashboardHelpers';
 import SchoolwideAttendance from './SchoolwideAttendance';
+import {apiFetchJson} from '../../../helpers/apiFetchJson';
+import measurePageLoad from '../../../helpers/measurePageLoad';
 
 class SchoolwideAbsences extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dashboardStudents: this.props.dashboardStudents,
+      isLoading: false
+    };
+  }
+
+  componentWillMount() {
+    //if the parent component already has the dashboard data, we don't need to duplicate it
```

Comment:
If I follow - this is because the existing `SchoolAdministratorDashboards` is calling this component (with the data read from `serializedData`), and now we're trying to call it from `App` but this component needs to get data on its own now.

You might consider either factoring this into a loader component that then calls `SchoolwideAbsences`, or moving the `DashboardHelpers.absenceEventsByDay` work into `SchoolwideAttendance`.  But also we'll remove the original caller shortly, so no use in overthinking this and just head towards whatever direction seems best :)

Reply:
I might end up using a less specific loader component for both pieces here or at least keeping the structure for the tardies loader and not doing more than one thing in this piece.